### DISCUSSION TITLE
Using with react

### DISCUSSION THREAD

**Discussion by LfJohnVo** [serversideup/docker-php#183] [Q&A] [answered]
Hi!!!! do you have any example to use this with laravel as api server and react in frontend? :D , i dont have any idea how to do this :C

**Comment by jaydrogers** [comment]
Open source, not yet... I hope to add some more examples in the future, but I am working on getting another open source project off the ground too ([Spin](https://github.com/serversideup/spin/pull/24))

Our book is a Nuxt frontend with a Laravel API. It's not react, but I assume it would be very similar. We have the source code available in the repo where it shows you how we're deploying everything with Docker Swarm with zero-downtime deployments on a server costing $5/mo.: 

The code is available in The Complete Package: https://serversideup.net/ultimate-guide-to-building-apis-and-spas-with-laravel-and-nuxt3/

I do want to emphasize the book does not cover the server deployment. The book itself is geared towards Nuxt & Laravel developers (not sysadmins)

**Reply by LfJohnVo** [reply]
Awesome!!! but docker swarm is now deprecated, still a good choice???

**Reply by jaydrogers** [reply, marked as answer]
There was a lot of confusion when the announcement came out. Swarm **Classic** is deprecated, not Swarm Mode: https://dockerlabs.collabnix.com/intermediate/swarm/difference-between-docker-swarm-vs-swarm-mode-vs-swarmkit.html

Swarm Mode is great if you want zero-downtime deployments without the headache of K8s. You can also run it on any machine for stupid cheap.

If you need [99.9999%](https://uptime.is/99.9999) uptime or more, then I would look into K8s.

I spent 500 hours learning K8s, to learn Swarm in 10 hours 😆
